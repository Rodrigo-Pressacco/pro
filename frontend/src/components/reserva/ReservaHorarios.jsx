import React, { useState } from "react";
import "../../styles/reserva/ReservaHorarios.css";

const ReservaHorarios = (props) => {

    const horarioPosible = ["10:00", "11:00", "12:00", "13:00", "14:00", 
                            "15:00", "16:00", "17:00",  "18:00",
                            "19:00", "20:00", "21:00",  "22:00", "23:00"]
    
    const [horaAEleccionShow, setHoraAEleccionShow] = useState(false);
    const [horaAEleccion, setHoraAEleccion] = useState("");

    const handleClick = (e) => {
        e.preventDefault()
        if (horaAEleccionShow) {
            setHoraAEleccionShow(false)
        }else{
            setHoraAEleccionShow(true)
        }
    }

    const handleClickDelHorario = (e) =>{
        e.preventDefault()
        setHoraAEleccion(e.target.textContent)
        props.horario(e.target.textContent)
        setHoraAEleccionShow(false)
    }

    const showHorarios = () => {
        if (horaAEleccionShow) {
            return(
                <ul className="reservas-ul">
                    {horarioPosible.map((e,i)=><li key={e} onClick={handleClickDelHorario}>{e}</li>)}
                </ul>
            )
            
        }
    }


    return (
        <div className="div-horarios">
            <p><i className="far fa-check-circle"></i> Tu habitaci√≥n va a estar lista para el check-in entre las 10:00 AM y las 11:00 PM</p>
            <label>Indique el horario de llegada:</label>
            <div className="show-horarios">
            <input type="text" name="" id="" onFocus={handleClick} value={horaAEleccion} readOnly placeholder="* Seleccionar hora"/>
            {showHorarios()}
            </div>

        </div>
    )
}
export default ReservaHorarios;