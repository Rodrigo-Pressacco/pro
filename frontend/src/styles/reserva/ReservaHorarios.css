.div-horarios {
    background-color: #fff;
    height: 144px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    border: 1px solid #dfe4ea;
    box-sizing: border-box;
    box-shadow: 0px 4px 4px rgb(0 0 0 / 25%);
    border-radius: 8px;
}

.div-horarios p,
.div-horarios label,
.div-horarios input {
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 14px;
    color: #383b58;
}

.div-horarios p {
    height: 20px;
    margin: 20px 0 10px 20px;
}

.div-horarios label {
    height: 20px;
    margin: 10px 0 10px 20px;
}

.div-horarios input {
    margin: 10px 0 20px 20px;
    height: 40px;
    width: 50%;
    box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
    border-radius: 5px;
    border: none;
    padding-left: 15px;
}

.reservas-ul {
    position: absolute;
    background-color: white;
    z-index: 1;
    top: 52px;
    left: 20px;
    width: 50%;
    border: 1px solid #dfe4ea;
    box-sizing: border-box;
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
    border-radius: 8px;
    padding: 5px 0;
}

.reservas-ul li {
    padding: 7px 0 7px 15px;

    font-weight: 500;
    font-size: 16px;
    line-height: 14px;
    color: #383b58;
}

.reservas-ul li:hover {
    background-color: #1dbeb35d;
}

.show-horarios {
    position: relative;
}

@media (max-width: 485px) {
    .div-horarios input {
        width: 90%;
    }
}
