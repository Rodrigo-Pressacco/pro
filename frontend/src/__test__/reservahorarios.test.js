import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import ReservaHorarios from '../components/reserva/ReservaHorarios'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<ReservaHorarios />', () => {
    let reservaHorarios;
    let handleClick = jest.fn(); 
    let handleClickDelHorario = jest.fn();
    let showHorarios = jest.fn();

    beforeEach(() => {
        reservaHorarios = shallow(<ReservaHorarios handleClick={handleClick} handleClickDelHorario={handleClickDelHorario} showHorarios={showHorarios} />)
    });

    it("Deberia mostrar <ReservaHorarios/> correctamente", () => {
        expect(reservaHorarios).toMatchSnapshot();
    });

    it("Deberia retornar 1 etiqueta p", () => {
        let node = reservaHorarios.find("p")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 1 etiqueta i", () => {
        let node = reservaHorarios.find("i")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 1 etiqueta label", () => {
        let node = reservaHorarios.find("label")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 1 etiqueta input", () => {
        let node = reservaHorarios.find("input")
        expect(node.length).toEqual(1)
    });

})