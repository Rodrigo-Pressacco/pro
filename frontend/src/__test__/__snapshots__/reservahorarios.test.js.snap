// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReservaHorarios /> Deberia mostrar <ReservaHorarios/> correctamente 1`] = `
<div
  className="div-horarios"
>
  <p>
    <i
      className="far fa-check-circle"
    />
     Tu habitaci√≥n va a estar lista para el check-in entre las 10:00 AM y las 11:00 PM
  </p>
  <label>
    Indique el horario de llegada:
  </label>
  <div
    className="show-horarios"
  >
    <input
      id=""
      name=""
      onFocus={[Function]}
      placeholder="* Seleccionar hora"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
</div>
`;
